docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,603 @@
+=== Mapping a runtime field
+You map runtime fields by adding a `runtime` section under the mapping
+definition. Within that section, you
+<<modules-scripting-using,define a Painless script>>, which has access to the
+entire context of a document, including the original `_source` and any mapped
+fields plus their values. At search time, the script runs and generates values
+for each scripted field that is required for the query.
+
+NOTE: You can define a runtime field in the mapping definition without a
+script. {es} will look in `_source` for a field with the same name as the
+runtime field and use values from that field at query time.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request, but also make the script results available anywhere
+in a search request.
+
+The script in the following request extracts the day of the week from the
+`@timestamp` field, which is defined as a `date` type:
+
+[source,console]
+----
+PUT /my-index
+{
+  "mappings": {
+    "runtime": { <1>
+      "day_of_week": {
+        "type": "keyword", <2>
+        "script": { <3>
+          "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+        }
+      }
+    },
+    "properties": {
+      "timestamp": {"type": "date"}
+    }
+  }
+}
+----
+
+<1> Runtime fields are defined in the `runtime` section of the mapping
+definition.
+<2> Each runtime field has its own field type, just like any other field.
+<3> The script defines the evaluation to calculate at search time.
+
+The `runtime` section supports `boolean`, `date`, `double`, `geo_point` `ip`,
+`keyword`, and `long` data types. Runtime fields with a `type` of `date` can
+accept the <<mapping-date-format,`format`>> parameter exactly as the `date`
+field type.
+
+You can define a runtime field that isn't at the top level of a document. If
+you want to map a runtime field under an object, you can use dot notation
+instead of recreating the entire object structure.
+
+For example, let's say you add the following document:
+
+[source, console]
+----
+POST my-index/_doc/1
+{
+  "company": {
+    "name": "Elastic"
+  }
+}
+----
+
+If you retrieve the mapping for that document, you'll see that the `name` field
+is nested under the `company` object:
+
+[source, console]
+----
+GET my-index/_mapping
+----
+//TEST[continued]
+
+[source, console-result]
+----
+{
+  "my-index" : {
+    "mappings" : {
+      "properties" : {
+        "company" : {
+          "properties" : {
+            "name" : {
+              "type" : "text",
+              "fields" : {
+                "keyword" : {
+                  "type" : "keyword",
+                  "ignore_above" : 256
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+----
+//TEST[continued]
+
+You could create a runtime field in the mapping using `company.name` instead of
+recreating the object structure for that field:
```

Comment:
After talking to Nik, I decided to remove this example and move this section under the Shadowing page, since we're effectively showing how to shadow a field that is part of an object.

Reply:
I agree on the confusion, but not on the solution :) I think we can mention how to map runtime fields that belong to objects independently of shadowing. The point of this example should be to show how to map company.name as a runtime field, rather than how to shadow the existing indexed field.

Reply:
Another idea, is to have runtime fields that belong to objects in most of our examples instead of at the top-level, and not mention the dot notation explicitly, because examples makes it clear how to do it.